File: 백준/Bronze/5585. 거스름돈/거스름돈.py
price = int(input())

count = 0
change = 0
rest = 1000 - price

while True:        
    if rest >= 500:
        change = 500
    elif rest >= 100:
        change = 100
    elif rest >= 50:
        change = 50
    elif rest >= 10:
        change = 10
    elif rest >= 5:
        change = 5
    elif rest >= 1:
        change = 1
    else: # rest == 0
        break
    
    rest -= change
    count += 1

print(count)

File: 백준/Bronze/5585. 거스름돈/거스름돈.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        
        int pay = sc.nextInt();
        int rest = 1000-pay;
        int count = 0;

        while(rest!=0) {
            if(rest>=500) {
                count++;
                rest -= 500;
            }
            else if(rest>=100) {
                count++;
                rest -= 100;
            }
            else if(rest>=50) {
                count++;
                rest -= 50;
            }
            else if(rest>=10) {
                count++;
                rest -= 10;
            }
            else if(rest>=5) {
                count++;
                rest -= 5;
            }
            else if(rest>=1) {
                count++;
                rest -= 1;
            }
        }

        System.out.println(count);
    }
}
